storage/gc_queue.go

```diff
@@ -150,7 +150,7 @@ func (gcq *gcQueue) process(now roachpb.Timestamp, repl *Replica,
 
 	snap := repl.store.Engine().NewSnapshot()
 	desc := repl.Desc()
-	iter := newReplicaDataIterator(desc, snap)
+	iter := newReplicaDataIterator(desc, snap, false /* !replicatedOnly */)
```

Comment:
The GC queue will never do anything with any unreplicated data, so we could pass true here and make the GC scan a bit less expensive (mainly by ignoring the raft log). However, if we did so the GC queue would no longer be a superset of the verification queue and it would no longer be appropriate to set the verification timestamp at the end of the GC queue's run (see #4710). 

I don't have a good enough sense of how often the GC queue runs to tell whether it's a net win to make GC scans cheaper at the expense of more frequent verification scans. However, if we leave this set to false it should at least be commented that we are doing so to ensure that this doubles as a verification scan. (Maybe `replicaDataIterator` should be responsible for setting the verification timestamp instead of the queues, so it can be set whenever a full iteration has completed but not for partial ones)